src/actions.ml

```diff
@@ -2288,3 +2288,86 @@ let coq_check_stale_pr ~bot_info ~owner ~repo ~after ~throttle =
     >>= fun () -> Lwt.return true
   in
   apply_after_label ~bot_info ~owner ~repo ~after ~label ~action ~throttle ()
+
+let run_bench ~bot_info comment_info =
+  (* Do we want to use this more often? *)
+  let open Lwt.Syntax in
+  let owner = comment_info.issue.issue.owner in
+  let repo = comment_info.issue.issue.repo in
+  let pr_number = comment_info.issue.number in
+  (* We need the GitLab build_id and project_id. Currently there is no good way
+     to query this data so we have to jump through some somewhat useless hoops in
+     order to get our hands on this information. TODO: do this more directly.*)
+  let* gitlab_check_summary =
+    GitHub_queries.get_pull_request_refs ~bot_info ~owner ~repo
+      ~number:pr_number
+    >>= function
+    | Error err ->
+        Lwt.return_error
+          (f
+             "Error while fetching PR refs for %s/%s#%d for running bench job: \
+              %s"
+             owner repo pr_number err )
+    | Ok {base= {sha= base}; head= {sha= head}} -> (
+        let base = Str.global_replace (Str.regexp {|"|}) "" base in
+        let head = Str.global_replace (Str.regexp {|"|}) "" head in
+        GitHub_queries.get_base_and_head_checks ~bot_info ~owner ~repo
+          ~pr_number ~base ~head
```

Comment:
For testing, this is OK, but for merging the feature, it would be preferable to add a custom query that allows you to get directly the checks for the head of the PR (and not the base).

Reply:
However `GitHub_queries.get_base_and_head_checks` is quite a large and complicated query. I am not really sure how to break it down into a smaller one.

Reply:
I can take care of that.